client/controller/index.web.js

```diff
@@ -387,3 +389,12 @@ export const notFound = ( context, next ) => {
 
 	next();
 };
+
+export const setSelectSiteIdByOrigin = ( context, next ) => {
```

Comment:
Nit: can we call this `setSelectedSiteIdByOrigin`?

Also, the query param could be `origin_site_id` to be very clear to future devs (it's an ID, not slug).